iso8601/iso8601.py

```diff
@@ -147,3 +147,16 @@ def parse_date(
         )
     except Exception as e:
         raise ParseError(e)
+
+
+def is_iso8601(datestring: str) -> bool:
+    """Check if a string matches an ISO 8601 format.
+
+    :param datestring: The string to check for validity
+    :returns: True if the string matches an ISO 8601 format, False otherwise
+    """
+    try:
+        m = ISO8601_REGEX.match(datestring)
+        return bool(m)
+    except Exception:
```

Comment:
My expectation is the regex matches the string or it doesn't, so anything else could be treated as a ParseError. 

i.e. I think we can change this to 

```python
except Exception as e:
    raise ParseError(e)
```

Reply:
That's fine with me! I was just following the convention established on the first few lines of the `parse_date()` function. Should that _also_ raise a `ParseError` in that case?